File: backend/electron-ecommerce-rest/src/main/java/com/electron/rest/dto/PageableResponse.java
package com.electron.rest.dto;

import lombok.*;

import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PageableResponse<T> {
    private String resourceName;
    private List<T> content;
    private int pageNo;
    private int pageSize;
    private int totalPages;
    private long totalElements;
}

File: frontend/electron-ecommerce-react/src/api/dto/PageableResponse.ts
export type PageableResponse<T> = {
    pageNo: number,
    totalPages: number,
    totalElements: number,
    pageSize: number,
    content: T[],
}

export const defaultPageableProduct = {
    pageNo: 0,
    totalPages: 0,
    totalElements: 0,
    pageSize: 25,
    content: [],
}